narwhals/_pandas_like/expr.py

```diff
@@ -128,13 +128,11 @@ def func(df: PandasLikeDataFrame) -> list[PandasLikeSeries]:
                     for column_name in evaluate_column_names(df)
                 ]
             except KeyError as e:
-                missing_columns = [
-                    x for x in evaluate_column_names(df) if x not in df.columns
-                ]
-                raise ColumnNotFoundError.from_missing_and_available_column_names(
-                    missing_columns=missing_columns,
-                    available_columns=df.columns,
-                ) from e
+                check_columns_exists(
+                    evaluate_column_names(df), available_columns=df.columns, from_error=e
+                )
+                msg = "Unreachable code, please report a bug."
+                raise AssertionError(msg) from e
```

Comment:
I had to add this otherwise the typechecker complains.

Can you think of a cleaner/better way?